traits/trait_types.py

```diff
@@ -2283,16 +2283,41 @@ def __init__(self, *types, **metadata):
             if len(types) == 0:
                 types = [Trait(element) for element in default_value]
 
-        self.types = tuple([trait_from(type) for type in types])
+        self.types = tuple(trait_from(type) for type in types)
         self.init_fast_validate(ValidateTrait.tuple, self.types)
 
         if default_value is None:
-            default_value = tuple(
-                [type.default_value()[1] for type in self.types]
+            # Optimisation: if all child traits have a constant default value,
```

Comment:
Hmm. This isn't actually purely optimisation: we don't validate defaults in the static case, so in current Traits we can have for example:

```
>>> from traits.api import *
>>> class A(HasTraits):
...     foo = Tuple(Int("some string"), Str(5))
... 
>>> a = A()
>>> a.foo
('some string', 5)
```

The "optimisation" here ensures that this behaviour (or arguably misbehaviour) persists.